src/corefx/System.Globalization.Native/localeStringData.cpp

```diff
@@ -158,6 +161,66 @@ UErrorCode GetLocaleIso3166CountryName(const char* locale, UChar* value, int32_t
 }
 
 /*
+Function:
+GetLocaleIso3166CountryCode
+
+Gets the 3 letter country code for a locale (via uloc_getISO3Country) and converts the result to UChars
+*/
+UErrorCode GetLocaleIso3166CountryCode(const char* locale, UChar* value, int32_t valueLength)
+{
+    const char *pIsoCountryName = uloc_getISO3Country(locale);
+    int len = strlen(pIsoCountryName);
+
+    if (len == 0)
+    {
+        return U_ILLEGAL_ARGUMENT_ERROR;
+    }
+
+    return u_charsToUChars_safe(pIsoCountryName, value, valueLength);
+}
+
+/*
+Function:
+GetLocaleCurrencyName
+
+Gets the locale currency English or native name and convert the result to UChars
+*/
+UErrorCode GetLocaleCurrencyName(const char* locale, bool nativeName, UChar* value, int32_t valueLength)
+{
+    UErrorCode status = U_ZERO_ERROR;
+    
+    UChar currencyThreeLettersName[4]; // 3 letters currency iso name + NULL
+    ucurr_forLocale(locale, currencyThreeLettersName, 4, &status);
+    if (!U_SUCCESS(status))
+    {
+        return status;
+    }
+    
+    int32_t len;
+    UBool formatChoice;
+    const UChar *pCurrencyLongName = ucurr_getName(
+                                        currencyThreeLettersName, 
+                                        nativeName ? locale : ULOC_US, 
+                                        UCURR_LONG_NAME, 
+                                        &formatChoice, 
+                                        &len, 
+                                        &status);
+    if (!U_SUCCESS(status))
+    {
+        return status;
+    }
+    
+    if (len >= valueLength) // we need to have room for NULL too
```

Comment:
u_charsToUChars_safe convert and copy from char* to UChar* but in my case I need just copy UChar* buffer to another UChar* buffer